### PR TITLE
[MBL-15353][Student][Teacher] Add Push notification settings 

### PR DESCRIPTION
**Pull request by hermannakos**
refs: MBL-15353
affects: Teacher
release note: Push notification settings can be changed from the app.

test plan: For both apps test the Settings > Push notifications screen. Check the api response

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: libs/pandautils/src/test/java/com/instructure/pandautils/features/notification/preferences/NotificationPreferencesViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+    private val apiPrefs: ApiPrefs = mockk(relaxed = true)
+    private val resources: Resources = mockk(relaxed = true)
+    private lateinit var notificationPreferenceUtils: NotificationPreferenceUtils
+
+    @Before
+    fun setUp() {
+        lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_CREATE)
+        Dispatchers.setMain(testDispatcher)
+
+        mockkStatic("kotlinx.coroutines.AwaitKt")
```
Comment (tamaskozmer) [inline libs/pandautils/src/test/java/com/instructure/pandautils/features/notification/preferences/NotificationPreferencesViewModelTest.kt:69]: We only need this, when we use the `awaitAll` call in the ViewModel.

**Review by tamaskozmer — CHANGES_REQUESTED** [review]

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by tamaskozmer — APPROVED** [review]

**Review by kdeakinstructure — APPROVED** [review]
QA 👍

**Pull request merged**